### ISSUE TITLE
Finetuning issue on V1

### ISSUE THREAD

**Issue by kushia** [SWivid/F5-TTS#848]
### Checks

- [x] This template is only for question, not feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I confirm that I am using English to submit this report in order to facilitate communication.

### Question details

Hi everyone,
Thanks for the impressive works on this !

I'm currently trying two trainings: 
- One from scratch with 700 hours of french (7 seconds average, std 3s)
- Same dataset but this time finetuning the Base model (Same dataset than above)

The training from scratch is ongoing so I don't have results yet but the loss seems ok at this time. (But any guidance on how much steps for training Base_v1 from scratch with 700h dataset would be appreciated. I saw 300k for small, but for me it didn't give good results after 400k on small. Hence why I'm trying base_v1)
But for the finetuning, the loss seemed to be good, the samples that have been generated with --log_samples are very good with impressive cloning capabilities (Even the accents).
But when I use it with f5-tts_infer_cli on a checkpoint (20 000 updates at grad_accum 4, so I assume 80 000 steps), it's very bad with a BIG english accent. (The only difference with the base weight is that it pronounce some rare liaisons Okay-ish)
I don't understand how the samples generated in the training are so great, yet, the actual generation are so bad.
Any help would be very appreciated!

Thanks!

**Comment by SWivid** [comment]
Hi @kushia , as mentioned in https://github.com/SWivid/F5-TTS/tree/main/src/f5_tts/train#2-finetuning-practice
> The `use_ema = True` is harmful for early-stage finetuned checkpoints (which goes just few updates, thus ema weights still dominated by pretrained ones), try turn it off and see if provide better results.

and the log_sample went without ema. https://github.com/SWivid/F5-TTS/blob/128f4e4bf39da70a4095e57e98aed694948ded45/src/f5_tts/model/trainer.py#L407

\
\
you could turn it off here: https://github.com/SWivid/F5-TTS/blob/128f4e4bf39da70a4095e57e98aed694948ded45/src/f5_tts/infer/utils_infer.py#L239
or just copy the model_state_dict to replace ema_model_state_dict in .pt / .safetensor checkpoint file

**Comment by SWivid** [comment]
> training Base_v1 from scratch with 700h dataset would be appreciated. I saw 300k for small, but for me it didn't give good results after 400k on small.

thought it could be with batchsize. if you use a much smaller batch size, it will take much longer to train.

**Comment by kushia** [comment]
Hi @SWivid 
Thanks for you answer!

Wow! The use_ema False is changing everything! You are a life saver sir! Thanks!

For the training from scratch, I'm using 2 H100, so same batch size than in the paper, with grad_accumulation at 1. (Wasn't sure it would really trigger any changes, and since the code logic is to use updates for checkpoints etc, and that steps == updates with grad acc at 1, I prefered to go on the easy way)

**Comment by kushia** [comment]
PS : Should I turn off use_ema also for the model trained from scratch ? Because the small was pretty good too on the samples, but not on infer .. (Shame I deleted the small checkpoints to train the base v1 so I can't test)
Is use_ema a legacy thing ?

**Comment by SWivid** [comment]
Yes for sure, sft is way suitable in this case.

For batch size, we use 8*80G gpu, so it would be a same batch_per_gpu setting and grad_accum to 4 if 2 H100.

> Should I turn off use_ema also for the model trained from scratch ? Is use_ema a legacy thing ?

At early stage, yes, cuz the model is not stable yet. And in a long round, use_ema=True is doing the Exponential Moving Average, which help improve stability of performance.

\ 
Wish good result, and have fun~

**Comment by kushia** [comment]
Sorry for all those questions, so if I understand correctly : 
For the training from scratch, I should use grad_accum to 4, but in this case, what am I monitoring to reproduce your number of steps ?
I thought that number of steps was invariant of grad accum. But in this case, for the scaling to be correct, updates == steps at grad_accum 4 with 2 h100 to reproduce your setting, is that correct ?

Thanks again!

**Comment by SWivid** [comment]
No worry.
It's OK now (since v0.4) to directly compare with same postfix, e.g. our `model_400000.pt` with your `model_400000.pt`.

The checkpoint saving logic is changed from per step (before v0.4) to per update, so the current model checkpoint is saved with actual loss bp-ed update (i.e. grad_accum already considered).
To be more clear, your `model_400000.pt` with `2 gpu * batchsize 38400 * grad_accum 4`, now equals to our `model_400000.pt` with `8 gpu * batchsize 38400 * grad_accum 1`.

**Comment by kushia** [comment]
Thank you again, it really helped! Have a nice day
